packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -18,21 +18,46 @@ import { assert, performance } from "@fluidframework/common-utils";
  */
 export const latencyThreshold = 5000;
 
+type IOpPerfTelemetryRequiredProperties =
+    /** Track how long op is sitting in inbound queue until it is processed */
+    "durationInboundQueue" |
+    /** Measure time outbound op is sitting in queue due to active batch */
+    "durationOutboundQueue" |
+    /** Time spent between DeltaManager's inbound "push" event until the DeltaManager "op" event */
+    "durationInboundToProcessing" |
+    /** Length of the DeltaManager's inbound queue at the time of the processing */
+    "lenghtInboundQueue";
+
+type IOpPerfTelemetryProperties =
+    Pick<ITelemetryProperties, IOpPerfTelemetryRequiredProperties>;
```

Comment:
I saw this technique used in summarizer, but I'm not sure I like it. I'd rather go with proper definition of interface, which would be strongly typed.